tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,82 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExtendedFxConversionService;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.*;
+import java.math.RoundingMode;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    private final ExternalQuotesService externalQuotesService;
+
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation,
+                                                   Beneficiary beneficiary,
+                                                   List<QuotePrice> quotePrices,
+                                                   BigDecimal amountOfRequest) {
+
+//new CompareQuotesBenificiary(operation, beneficiary)
+        Optional<QuotePrice> quotePrice = findQuote(new CompareQutesExtend(),
+                quotePrices,
+                amountOfRequest);
+
+        if (!quotePrice.isPresent()) {
+            return Optional.empty();
+        }
+        return Optional.of(quotePrice.get().getPricePerPiece());
+    }
+
+    Optional<QuotePrice> findQuote(Comparator<QuotePrice> comparator,
+                              List<QuotePrice> quotePrices,
+                              BigDecimal amountOfRequest) {
+        BigDecimal priceInCurr;
+        List<QuotePrice> finalQuoteList = new ArrayList<>();
+        for (QuotePrice quotePrice: quotePrices) {
+            priceInCurr  = amountOfRequest.divide(quotePrice.getPrice(), 10, RoundingMode.HALF_UP);
+            if (quotePrice.getAmount().compareTo(priceInCurr) > 0
+            || quotePrice.getAmount().compareTo(BigDecimal.ZERO) < 0) {
```

Comment:
&#39;if&#39; child has incorrect indentation level 12, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a608855745>*